backend/src/apis/round.rs

```diff
@@ -75,100 +85,119 @@ where
-                    .collect::<Vec<Cryptocurrency>>()
-                    .as_slice()
-                    .try_into()
-                    .unwrap(),
-                U256::from(self.get_timestamp()),
-            )
-            .await?;
-
         Ok(())
     }
 
     pub fn get_proof_of_inclusion(
         &self,
         user_index: usize,
-    ) -> Result<MstInclusionProof, &'static str>
+    ) -> Result<KZGInclusionProof, &'static str>
     where
-        [(); N_CURRENCIES + 2]: Sized,
+        [(); N_CURRENCIES + 1]: Sized,
     {
+        // Iterate unblinded advice polynomials evaluate balances array
         Ok(self
             .snapshot
             .generate_proof_of_inclusion(user_index)
             .unwrap())
     }
 }
 
-impl<const LEVELS: usize, const N_CURRENCIES: usize, const N_BYTES: usize>
-    Snapshot<LEVELS, N_CURRENCIES, N_BYTES>
+/// The `Snapshot` struct represents the state of database that contains users balance on holds by Custodians at a specific moment.
+///
+/// # Fields
+///
+/// * `advice_polys`: Composed of the unblinded advice polynomial, `advice_poly`, and the polynomials of blind factors, `advice_blind`.
+/// * `user_balances`: A 2D array of user identity and balances.
+/// * `trusted_setup`: The trusted setup artifacts generated from the `UnivariateGrandSum` circuit.
+///
+pub struct Snapshot<const N_CURRENCIES: usize, const N_POINTS: usize, const N_USERS: usize> {
+    advice_polys: AdviceSingle<G1Affine, Coeff>,
+    trusted_setup: (
+        ParamsKZG<Bn256>,
+        ProvingKey<G1Affine>,
+        VerifyingKey<G1Affine>,
+    ),
+}
+
+impl<const N_CURRENCIES: usize, const N_POINTS: usize, const N_USERS: usize>
+    Snapshot<N_CURRENCIES, N_POINTS, N_USERS>
 where
     [usize; N_CURRENCIES + 1]: Sized,
-    [usize; N_CURRENCIES + 2]: Sized,
 {
     pub fn new(
-        mst: Box<dyn Tree<N_CURRENCIES, N_BYTES>>,
+        advice_polys: AdviceSingle<G1Affine, Coeff>,
         params_path: &str,
-    ) -> Result<Snapshot<LEVELS, N_CURRENCIES, N_BYTES>, Box<dyn std::error::Error>> {
-        let mst_inclusion_circuit = MstInclusionCircuit::<LEVELS, N_CURRENCIES, N_BYTES>::init_empty();
+    ) -> Result<Snapshot<N_CURRENCIES, N_POINTS, N_USERS>, Box<dyn Error>> {
+        let univariate_grand_sum_circuit: UnivariateGrandSum<N_USERS, N_CURRENCIES> =
+            UnivariateGrandSum::<N_USERS, N_CURRENCIES>::init_empty();
 
         // get k from ptau file name
-        let parts: Vec<&str> = params_path.split("-").collect();
+        let parts: Vec<&str> = params_path.split('-').collect();
         let last_part = parts.last().unwrap();
         let k = last_part.parse::<u32>().unwrap();
 
-        let mst_inclusion_setup_artifacts: SetupArtifacts =
-            generate_setup_artifacts(k, Some(params_path), mst_inclusion_circuit).unwrap();
+        let univariant_grand_sum_setup_artifcats =
+            generate_setup_artifacts(k, Some(params_path), &univariate_grand_sum_circuit).unwrap();
```

Comment:
If we already have the polynomials at this step, it means that we have obtained the range check zkSNARK proof that used some setup, I think we can reuse the same setup here by also passing it as an argument.

Reply:
Oh, yes! you are right. I will fix soon.